src/opers.h

```diff
@@ -27,78 +63,155 @@ extern Obj TRY_NEXT_METHOD;
-    (TNUM_OBJ(func) == T_FUNCTION && SIZE_OBJ(func) == SIZE_OPER )
+static inline Int IS_OPERATION(Obj func)
+{
+    return TNUM_OBJ(func) == T_FUNCTION && SIZE_OBJ(func) == sizeof(OperBag);
+}
+
+
+/****************************************************************************
+**
+*F  OPER
+*/
+static inline OperBag * OPER(Obj oper)
+{
+    GAP_ASSERT(IS_OPERATION(oper));
+    return (OperBag *)ADDR_OBJ(oper);
+}
+
+static inline const OperBag * CONST_OPER(Obj oper)
+{
+    GAP_ASSERT(IS_OPERATION(oper));
+    return (const OperBag *)CONST_ADDR_OBJ(oper);
+}
+
 
 /****************************************************************************
 **
 *F  FLAG1_FILT( <oper> )  . . . . . . . . . .  flag 1 list of an `and' filter
 */
-#define FLAG1_FILT(oper)        (*            (ADDR_OBJ(oper) +16     ) )
+static inline Obj FLAG1_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->flag1;
+}
+
+static inline void SET_FLAG1_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flag1 = x;
+}
 
 
 /****************************************************************************
 **
 *F  FLAG2_FILT( <oper> )  . . . . . . . . . .  flag 2 list of an `and' filter
 */
-#define FLAG2_FILT(oper)        (*            (ADDR_OBJ(oper) +17     ) )
+static inline Obj FLAG2_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->flag2;
+}
+
+static inline void SET_FLAG2_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flag2 = x;
+}
 
 
 /****************************************************************************
 **
 *F  FLAGS_FILT( <oper> )  . . . . . . . . . . . . . . . . . flags of a filter
 */
-#define FLAGS_FILT(oper)        (*            (ADDR_OBJ(oper) +18     ) )
+static inline Obj FLAGS_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->flags;
+}
+
+static inline void SET_FLAGS_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flags = x;
+}
 
 
 /****************************************************************************
 **
 *F  SETTER_FILT( <oper> ) . . . . . . . . . . . . . . . .  setter of a filter
 */
-#define SETTR_FILT(oper)        (*            (ADDR_OBJ(oper) +19     ) )
+static inline Obj SETTR_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->setter;
+}
+
+static inline void SET_SETTR_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->setter = x;
+}
 
 
 /****************************************************************************
 **
 *F  TESTR_FILT( <oper> )  . . . . . . . . . . . . . . . .  tester of a filter
 */
-#define TESTR_FILT(oper)        (*            (ADDR_OBJ(oper) +20     ) )
+static inline Obj TESTR_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->tester;
+}
+
+static inline void SET_TESTR_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->tester = x;
+}
 
 
 /****************************************************************************
 **
 *F  METHS_OPER( <oper> )  . . . . . . . . . . . . method list of an operation
 */
-#define METHS_OPER(oper,i)      (*            (ADDR_OBJ(oper) +21+ (i)) )
+static inline Obj METHS_OPER(Obj oper, Int i)
+{
+    return CONST_OPER(oper)->methods[i];
```

Comment:
Could have bounds checks here and below in cache_oper

Reply:
Added the bounds checks.